experimental/nodejs-kafka/README.md

```diff
@@ -0,0 +1,88 @@
+
+The [appmetrics-prometheus](https://github.com/CloudNativeJS/appmetrics-prometheus) module will collect a wide range of resource-centric (CPU, memory) and application-centric (HTTP request responsiveness) metrics from your application, and then expose them as multi-dimensional time-series data through an application endpoint for Prometheus to scrape and aggregate.
+
+This stack also comes with Prometheus metrics, which has been preconfigured to work with your application. You will not be able to override this endpoint:
+
+- Metrics endpoint: http://localhost:3000/metrics
+
+## Templates
+
+Templates are used to create your local project and start your development. This stack provides one template, "default".
+
+The default template provides you with a basic Kafka producer and consumer and a sample test in a test sub directory.
+
+The application must have a top-level package.json file, and it must contain a property `"main": APP` (the templates all set `APP` to `"app.js"` and include a skeleton implementation in that file). The `APP` will be required, and must export a function.
+
+The function will be called with an `options` argument, which may be ignored.
+
+Options contains:
+- `Kafka` {node-rdkafka} Use this to configure and use node-rdkafka.
+- `log` {pino} an instance of [`pino`](https://www.npmjs.com/package/pino). It can be used for application specific logging. The default log level is `'info'` in production, and `'debug'` in non-production.
+- `app` {express} an instance of [`express`](https://expressjs.com), it may be used to setup routes, or it may be ignored if the service does not expose HTTP endpoints.
+
+## Getting Started
+
+1. Create a new folder in your local directory and initialize it using the Appsody CLI, e.g.:
+
+    ```bash
+    mkdir my-project
+    cd my-project
+    appsody init nodejs-kafka
+    ```
+
+    This will initialize a Node.js Kafka project using the default template.
+
+1. After your project has been initialized you can then run your application using the following command:
+
+    ```bash
+    appsody run
+    ```
+
+    This launches a Docker container that continuously re-builds and re-runs your project, exposing it on port 3000.
+
+    You can continue to edit the application in your preferred IDE (VSCode or others) and your changes will be reflected in the running container within a few seconds.
+
+1. You should be able to access the following endpoints, as they are exposed by your template application by default:
+
+    - Health endpoint: http://localhost:3000/health
+    - Liveness endpoint: http://localhost:3000/live
+    - Readiness endpoint: http://localhost:3000/ready
+    - Metrics endpoint: http://localhost:3000/metrics
+
+1.  A message may be posted into the default skeleton app. This could be done with `curl`:
+    ```bash
+    curl -v -X POST -H "content-type: application/json" \
```

Comment:
Running this command at first succeeds for me although eventually it looks like connection to the producer/consumer is lost:
```
[Container] {"level":50,"time":1584971907527,"pid":45,"hostname":"f4395b2e3ac3","msg":"error from consumer: Error: broker transport failure","v":1}
[Container] {"level":50,"time":1584971907527,"pid":45,"hostname":"f4395b2e3ac3","msg":"error from consumer: Error: all broker connections are down","v":1}
```
which then results in failures running this. 

Also nit: You could cut out the use of the -X flag. Not a big deal but I get this note when running the command:
```
Note: Unnecessary use of -X or --request, POST is already inferred.
```

Reply:
I see this problem too.  Are you running the https://github.com/lensesio/fast-data-dev image that Sam suggested?  I also tried using a `docker-compose.yaml` for Kafka that I had lying around, but that has the same problem.